python/cuml/dask/datasets/regression.py

```diff
@@ -179,59 +274,99 @@ def make_regression(n_samples=100, n_features=100, n_informative=10,
+            X = f_order_standard_normal(client, rs, n_samples, n_features,
+                                        [n_samples_per_part] * n_parts, dtype)
+        elif order == 'C':
+            X = rs.standard_normal((n_samples, n_features),
+                                   chunks=(n_samples_per_part, (
+                                                                n_informative,
+                                                                n_features -
+                                                                n_informative)
+                                           ),
+                                   dtype=dtype)
 
     else:
         # Randomly generate a low rank, fat tail input set
-        X = make_low_rank_matrix(n_samples=n_samples,
+        X = make_low_rank_matrix(client=client,
+                                 n_samples=n_samples,
                                  n_features=n_features,
                                  effective_rank=effective_rank,
                                  tail_strength=tail_strength,
                                  random_state=rs,
-                                 n_parts=n_parts)
+                                 n_parts=n_parts,
+                                 dtype=dtype,
+                                 order=order)
         X = X.rechunk({0: n_samples_per_part,
-                       1: (n_informative, n_features-n_informative)})
+                      1: (n_informative, n_features-n_informative)})
 
     # Generate a ground truth model with only n_informative features being non
     # zeros (the other features are not correlated to y and should be ignored
     # by a sparsifying regularizers such as L1 or elastic net)
+    if effective_rank and not use_full_low_rank:
+        _, _, coef_ = sg_make_regression(n_samples=n_samples_per_part,
+                                         n_features=n_features,
+                                         n_informative=n_informative,
+                                         n_targets=n_targets,
+                                         bias=bias,
+                                         effective_rank=effective_rank,
+                                         tail_strength=tail_strength,
+                                         noise=noise,
+                                         shuffle=shuffle,
+                                         coef=True,
+                                         random_state=random_state,
+                                         dtype='double')
+        coef_ = cp.array(coef_, dtype=dtype)
+        ground_truth = da.from_array(coef_, chunks=(n_samples_per_part, -1))
+        y = da.dot(X, ground_truth) + bias
+    else:
+        ground_truth = 100.0 * rs.standard_normal((n_informative, n_targets),
+                                                  chunks=(n_samples_per_part,
+                                                          -1),
+                                                  dtype=dtype)
```

Comment:
Dask's make regression uses the hack of generating a low rank on a smaller chunk. Calling down to their code instead of cuml SG `make_regression` will cause us to train coefs on an even smaller chunk, if I am not wrong